implementations/rust/ockam/ockam_identity/src/identity.rs

```diff
@@ -5,31 +5,32 @@ use crate::change::IdentitySignedChange;
 use crate::change_history::{IdentityChangeHistory, IdentityHistoryComparison};
 use crate::credential::Credential;
 use crate::{
-    ChangeIdentifier, IdentityError, IdentityIdentifier, IdentityVault, KeyAttributes,
+    to_hasher, ChangeIdentifier, IdentityError, IdentityIdentifier, IdentityVault, KeyAttributes,
     PublicIdentity, SecureChannelRegistry,
 };
 use ockam_core::compat::{boxed::Box, string::String, sync::Arc, vec::Vec};
 use ockam_core::vault::Secret::Key;
 use ockam_core::vault::{
-    SecretKey, SecretPersistence, SecretType, Signature, CURVE25519_SECRET_LENGTH_U32,
+    SecretKey, SecretPersistence, SecretType, SecretVault, Signature, CURVE25519_SECRET_LENGTH_U32,
 };
 use ockam_core::{Address, Result};
 use ockam_core::{AsyncTryClone, DenyAll};
 use ockam_node::compat::asynchronous::RwLock;
 use ockam_node::Context;
+use ockam_vault::{Hasher, Vault};
 use ockam_vault::{KeyId, SecretAttributes};
 
 /// Identity implementation
 #[derive(AsyncTryClone)]
```

Comment:
Oh, actually it can't because of the Context...

Reply:
It's a little disappointing that we need to wrap Identity in an Arc. I think we should consider alternatives:
1. Wrap `Context` inside `Identity` in an `Arc`
2. Remove `Context` from `Identity` and derive `Clone` for `Identity` instead of `AsyncTryClone`

Reply:
I will try to tackle this with my next PR